posts/tests.py

```diff
@@ -109,4 +112,42 @@ def test_no_rendering_unsafe_tags(self):
         self.assertContains(self.response, '<h1>This is the main heading </h1> &lt;script&gt;alert("Hello");&lt;/script&gt;<h2>This is the sub heading </h2>', status_code=200)
 
     def test_to_check_Headings_in_outline(self):
-        self.assertContains(self.response, '<ul><li>This is the main heading  </li></ul><ul><ul><li>This is the sub heading  </li></ul></ul>', status_code=200)
\ No newline at end of file
+        self.assertContains(self.response, '<ul><li>This is the main heading  </li></ul><ul><ul><li>This is the sub heading  </li></ul></ul>', status_code=200)
+
+class LoggedInUserTesting(TestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(username='joel', password='123aaa789')
+
+    def test_user_login(self):
+        request = self.factory.get('/customer/details')
+        request.user = self.user
+        self.assertTrue(request.user)
+
+class SignUpLogInTest(TestCase):
+    def setUp(self):
+        self.signup1 = {'username': 'testuser2' , 'password1' : 'aaaa1234', 'password2' : 'aaaa1234'}
+        self.signup2 = {'username': 'testuser' , 'password1' : 'aaaaaaaa', 'password2' : 'aaaacccc'}
+        self.login = {'username': 'testuser','password': 'secret'}
+        self.login2 = {'username': 'testuser','password': 'secretsssss'}
+        User.objects.create_user(**self.login)
+
+    def test_signup(self):
+        response = self.client.post(reverse('signup'), self.signup1, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        self.assertRedirects(response, reverse('posts:create'))
+
+    def test_signup_wrong_password(self):
+        response = self.client.post(reverse('signup'), self.signup2, follow=True)
+        self.assertFalse(response.context['user'].is_active)
+        import pdb;pdb.set_trace()
```

Comment:
remove the debugger. If you are trying to test signups with different passwords, you should assert error message in response